test-suite/shared/src/test/scala/org/scalajs/testsuite/compiler/RegressionTest.scala

```diff
@@ -332,6 +332,26 @@ class RegressionTest {
     assertEquals(3, test(null))
   }
 
+  @Test def IR_checker_must_not_check_method_signatures_on_classes_with_no_instance(): Unit = {
+    assumeTrue("linking only", false)
+
+    class Foo // this class will be dropped by base linking
+
+    class Bar {
+      /* This method is called, but unreachable because there are no instances
+       * of `Bar`. It will therefore not make `Foo` reachable.
+       */
+      def meth(foo: Foo): String = foo.toString()
+    }
+
+    @noinline def nullBar(): Bar = null
+
+    // the IR checker must not try to infer the signature of these calls
+    nullBar().meth(null)
+    (null: Bar).meth(null)
+    (??? : Bar).meth(null) // scalastyle:ignore
+  }
+
```

Comment:
Could you also add a unit test for this?

Reply:
Er, isn't *that* a unit test? Do you mean a test in `linker/shared/test`?

Reply:
Yes, I meant a test in `linker/shared/test`? Unclear whether the test-suite is unit tests, it really depends on your definition of unit (but doesn't really matter, I meant to ask for a linker test :)).

Reply:
OK. I'll do that when I come back from vacation, probably, because it's going to be annoying to write on mobile ^^

Reply:
Done.